17.md

```diff
@@ -0,0 +1,111 @@
+NIP-17
+======
+
+Git Commits and Discovery Over Nostr
+------------------------------------
+
+`draft` `optional` `author:melvincarvalho`
+
+- Reserves:  `kind 17`
+
+Do not merge until implemented, work in progress
+------------------------------------------------
+
+Status: This NIP is an early draft and work in progress.  Implementations are being developed which will incorporate what is learnt.
+
+
+Introduction
+------------
+
+Git Commits over nostr enable **uses cases** similar to:
+
+- [github actions](https://github.com/features/actions)
+- [continuous integration](https://en.wikipedia.org/wiki/Continuous_integration)
+- [continuous deplyment](https://en.wikipedia.org/wiki/Continuous_deployment)
+- [travis](https://travis-ci.org/)
+
+but without central points of failure.  The aim is to create continuous distributed software development processes, without having to rely on complex or centralized services.
```

Comment:
The nip only describes how commit messages could be replicated on nostr but doesn't go into any of these other aspects. Is it really worth thousands of events with all the tiniest commit messages to convey the existence of a project?

Reply:
May expand on that, but I Think implementation and app will be more instructive.  Thanks for food for thought.